prefect_email/credentials.py

```diff
@@ -62,10 +64,9 @@ def _cast_to_enum(obj: Union[str, SMTPType], enum: Enum, restrict: bool = False)
         return getattr(enum, obj.upper())
 
 
-@dataclass
-class EmailCredentials:
+class EmailCredentials(Block):
```

Comment:
I think we should take this opportunity to rename this `EmailServerCredentials` to better describe what it encapsulates.
```suggestion
class EmailServerCredentials(Block):
```